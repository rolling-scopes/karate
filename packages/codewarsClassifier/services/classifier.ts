import { marksMapping, targetKatas } from '../marks'

const getMarkByRank = (rank: string): number => {
  return marksMapping[rank] || 0
}

export const classify = (katas: Array<any>) => {
  return katas.map(rankMapper).reduce((points: number, point: number) => points + point)
}

function rankMapper(kata: any) {
  const kataMapped = targetKatas.find(kn => kn.id === kata.id)
  return kataMapped ? getMarkByRank(kata.rank) : 0
}