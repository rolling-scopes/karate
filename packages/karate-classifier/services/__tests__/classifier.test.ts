import { classify } from '../classifier'
const mock = require('./mock.json')

test('Classifier', function () {
  const res = classify(mock)
  expect(res).toMatchSnapshot();
})