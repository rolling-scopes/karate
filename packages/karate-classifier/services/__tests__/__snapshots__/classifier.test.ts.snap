// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Classifier 1`] = `145`;
